File: frontend/src/types/book.ts
export interface Book {
    id?: number;
    title: string;
    author: string;
    user_id?: number;
    created_at?: string;
    updated_at?: string;
    deleted_at?: string | null;  // Optional since it's marked with json:"-" in Go
    completed: boolean;
} 

File: backend/internal/models/book.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Book struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	Title     string         `json:"title"`
	Author    string         `json:"author"`
	UserID    uint           `json:"user_id" gorm:"index"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`
	Completed bool           `json:"completed"`
}
